models/match.py

```diff
@@ -60,13 +60,20 @@ class Match(db.Document):
         "queryset_class": CustomQuerySet
     }
 
+    def clean(self):
```

Comment:
This approach is only clever, if we are sure there is no invalid data in the db already, which the system needs to be able to save _again_. Reading shouldn't be an issue, though.